webpack.config.js

```diff
@@ -0,0 +1,53 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const {CleanWebpackPlugin} = require('clean-webpack-plugin');
+const webpack = require('webpack');
+
+module.exports = {
+  entry: {
+    main: path.resolve(__dirname, '/client/.src/index.js')
+  },
+  output: {
+    path: path.resolve(__dirname, './public'),
+    filename: 'bundle.js'
+  },
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        use: ['babel-loader']
+      },
+      {
+        test: /\.(?:ico|gif|png|jpg|jpeg)$/i,
+        type: 'asset/resource'
+      },
+      {
+        test: /\.(woff(2)?|eot|ttf|otf|svg|)$/,
+        type: 'asset/inline'
+      },
+      {
+        test: /\.(scss|css)$/,
+        use: ['style-loader', 'css-loader', 'postcss-loader']
```

Comment:
more or less, just asking this for my own personal understanding.. what are these tests for beyond the first one with babel-loader? I haven't seen this done before, so I'm just curious.